src/kixi/heimdall/service.clj

```diff
@@ -120,48 +120,54 @@
     (fail "Invalid or expired refresh token provided")))
 
 (defn create-group-event
-  [db communications {:keys [group user-id] :as input}]
-  (let [group-ok? (and (spec/valid? :kixi.heimdall.schema/group-params group)
-                       (not (group/find-by-name db (:group-name group))))
-        user-ok? (spec/valid? :kixi.heimdall.schema/id user-id)]
+  [db communications {:keys [group-name group-id user-id] :as input}]
+  (let [group-ok? (and (spec/valid? :kixi.heimdall.schema/group-params input)
+                       (not (group/find-by-name db group-name)))
+        user-ok? (spec/valid? :kixi.heimdall.schema/id user-id)
+        input-dated (assoc input
+                           :created (str (util/db-now)))]
     (if (and user-ok? group-ok?)
-      (do (comms/send-event! communications :kixi.heimdall/group-created "1.0.0" input) true)
-      (do (comms/send-event! communications :kixi.heimdall/create-group-failed "1.0.0" {:user-valid user-ok?
+      (do (comms/send-event! communications :kixi.heimdall/group-created "2.0.0" input-dated) true)
+      (do (comms/send-event! communications :kixi.heimdall/create-group-failed "2.0.0" {:user-valid user-ok?
```

Comment:
This is quite small, so probably not an issue, but I generally prefer to separate data structure construction (the event and the `:kixi.heimdall/<event-key>` in this case) from the side effecting bit. Then it could be tested w/o needing integration.